### ISSUE TITLE
Locales and units

### ISSUE THREAD

**Issue by franglais125** [franglais125/gnome-shell-system-monitor-applet#2]
@DevDef thanks for reporting this issue https://github.com/paradoxxxzero/gnome-shell-system-monitor-applet/issues/359

I'll try to see how I can fix the remaining bits. I moved the conversation over here not to spam everyone with these small developments.

1. Je voulais d'abord confirmer qu'on a le même problème. Par exemple, vous avez dit que "Mbps" et "kbps" n'étaient pas traduits. Où est-ce que vous voyez ça? Voici mon indicateur:
![screenshot from 2017-04-29 19-54-47](https://cloud.githubusercontent.com/assets/19195975/25559989/290900ce-2d16-11e7-93cb-9a5bec161120.png)

2. Deuxièmement, il y a toujours un problème avec certains mots qui ne sont pas traduits, comme "buffer", etc. En voici un example:
![screenshot from 2017-04-29 19-57-21](https://cloud.githubusercontent.com/assets/19195975/25559990/2be8be7e-2d16-11e7-80c1-0fd0157b9d38.png)

Est-ce que dans les deux cas vous vouez la même chose que moi ?

**Comment by DevDef** [comment]
1. Je suppose que Mbps et kbps sont censés s’afficher lorsque l’on coche la case « Afficher le débit réseau en bits » dans l’onglet _Réseau_. Mais, étrangement, cela ne semble pas fonctionner.

2. C’est étrange, la version que j’utilise actuellement (venant du site extensions.gnome.org) affiche _Tampon_, _Cache_ et _Programme_.

**Comment by franglais125** [comment]
1. Ah! Bien vu, je n'avais pas remarqué ça!
2. En effet, j'avais mal installé les changements que j'avais apportés. Tampon, cache, etc. sont bien affichés.

**Comment by franglais125** [comment]
Donc, il faudrait changer "kbps" pour "kbit/s", et Mbps par "Mbit/s".

Est-ce qu'il faut aussi traduire ces unités? La page en français montre les mêmes noms qu'en anglais https://fr.wikipedia.org/wiki/D%C3%A9bit_binaire

**Comment by DevDef** [comment]
Oui, ce sont les unités officielles du système international (hérité de la révolution française \o/).
Donc, nul besoin de le traduire en français, mais j’ai supposé que ça pouvait s’écrire différemment dans d’autres langues, ce qui justifierait de « gettextizer » ces chaînes de caractères…

**Comment by franglais125** [comment]
Parfait, merci! Je vais proposer un patch pour ça.

**Comment by franglais125** [comment]
Bonjour encore!

Merci pour le nouveau PR que vous avez commencé, j'aurais dû tester mon code sur Gnome 3.24! J'étais en train de voyager et donc je n'ai vu tout ceci que maintenant.

Dans le PR que vous avez ouvert, il y a des erreurs de forme en ce moment. Si ça ne vous dérange pas, je vais l'intégrer dans le PR que j'avais déjà commencé et corriger les erreurs. Comme ça ce sera un peu plus condensé (donc moins de commits) et ordonné. Est-ce que ça vous va ?

**Comment by DevDef** [comment]
Salut,

Pas de problème, ça me va.
Pour ma culture et afin que je ne les reproduise pas, quelles sont ces erreurs de forme ?

P.-S. : j’aimerais que nous nous tutoyions, si cela ne te dérange pas.

**Comment by franglais125** [comment]
Ah, oui absolument, ça marche! On m'avait déjà dit que les Québécois tutoyaient trop, et j'essaie de faire attention, heh.

Quant aux erreurs, si tu suis le lien "Details", où ça dit qu'il y a des erreurs, tu vas les voir décrits: https://travis-ci.org/paradoxxxzero/gnome-shell-system-monitor-applet/builds/230070053?utm_source=github_status&utm_medium=notification

- Ligne 183: des espaces à la fin de la ligne
- Ligne 1857: le `if` n'a pas de `{}`
- Ligne 1857: utilisation de `!=` à la place de `!==`

Les trois ne sont pas du tout des problèmes majeurs, c'est juste qu'on venait d'embellir le code partout pour le rendre plus uniforme.

**Comment by DevDef** [comment]
Tu es de la Belle Province ! Salut à toi cousin des Amériques.
En ce qui concerne le tutoiement, il me semble qu’il est d’usage dans la communauté libriste, comme dans toute communauté d’ailleurs…

Venons-en à mes erreurs :
  - pour les espaces en fin de ligne, c’est une chose qui en principe ne m’échappe pas, tout comme les horribles mélanges d’indentations avec des espaces et des tabulations, car j’active l’affichage des caractères invisibles dans mes éditeurs (Gedit, Eclipse, Netbeans…), mais dans le cas présent j’ai utilisé l’édition en ligne de GitHub ;
  - il est vrai qu’il est recommandé d’utiliser systématiquement des accolades après un `if`, je transgresse parfois cette bonne pratique ;
  - quant au `!==` à la place du `!=`, c’est vrai que c’est mieux.

J’attends à présent avec impatience l’intégration de ton correctif d’alignement des valeurs avec leurs unités…

Enfin, j’ai une petite suggestion concernant le `Makefile`. Il serait appréciable pour les empaqueteurs d’avoir la possibilité de configurer les emplacements du fichier schéma (sans le fichier compilé) et des traductions, respectivement dans `/usr/share/glib-2.0/schemas/` et `/usr/share/locale/`.
L’extension [gnome-shell-extension-openweather](https://github.com/jenslody/gnome-shell-extension-openweather) y parvient grâce aux Autotools (voir [`autogen.sh`](https://github.com/jenslody/gnome-shell-extension-openweather/blob/master/autogen.sh), [`configure.ac`](https://github.com/jenslody/gnome-shell-extension-openweather/blob/master/configure.ac) et [`Makefile.am`](https://github.com/jenslody/gnome-shell-extension-openweather/blob/master/Makefile.am)), ce qui permet d’avoir un [fichier SPEC simplifié](https://github.com/jenslody/gnome-shell-extension-openweather/blob/master/gnome-shell-extension-openweather.spec) dans lequel la construction se résume (après `NOCONFIGURE=1 ./autogen.sh`) au traditionnel triptyque :
```bash
./configure [options]
make
make install
```

Librement,

Davy

P.-S. : Je cherche toujours un moyen de traduire la description de l’extension qui s’affiche dans GNOME Tweak tools et qui se trouve dans le fichier metadata.json. La toute fin de [_ce fichier `fr.po`_](https://github.com/mockturtl/cinnamon-weather/blob/master/po/fr.po) me laisse supposer que la chose est réalisable. Malheureusement, la méthode d’extraction des chaînes à traduire n’est pas indiquée…
Je vais tout de même vérifier qu’en ajoutant manuellement ladite chaîne à traduire à mon fichier `fr.po` et après recompilation, la description est ensuite traduite.

**Comment by DevDef** [comment]
> Je vais tout de même vérifier qu’en ajoutant manuellement ladite chaîne à traduire à mon fichier fr.po et après recompilation, la description est ensuite traduite.

Essai non concluant, hélas !

**Comment by franglais125** [comment]
Ah! Dommage! Désolé, dernièrement j'étais en train de travailler sur d'autres extensions aussi! Et je n'ai pas eu de réponse encore sur le PR que j'avais ouvert.

Merci pour toute l'information apportée!

**Comment by DevDef** [comment]
Salut,

La ferveur semble être retombée dirait-on. C’est dommage tes dernières propositions m’enthousiasmaient. :-/

**Comment by franglais125** [comment]
Bonjour!

Effectivement, je ne comprends pas le manque d'attention. Le gros du travail est fait, il ne suffirait que de me faire savoir s'il y a des problèmes...

J'imagine que quand Gnome 3.26 va sortir, il va falloir au moins mettre le metadata à jour, et on verra là peut-être des nouveautés.

**Comment by DevDef** [comment]
OK, merci de ton retour qui, lui, était on ne peut plus réactif. :-)
